CDCgov/prime-reportstream
# added RADx Mars OTC Onboarding conformance profile

This PR ...

Adds RADx Mars OTC Onboarding conformance profile and topic.

Test Steps:
1. Send prime-router/src/test/resources/validation/marsotcelr/sample_3.hl7 as a sender using the mars-otc-elr topic.
2. Check the history API using the report ID from step 1 to ensure there were no errors.
3. Send prime-router/src/test/resources/validation/marsotcelr/sample_3.hl7 as a sender using the mars-otc-elr-onboarding topic.
4. Check the history API using the report ID from step 3 to ensure that there were errors.

To perform the test steps locally using curl:
1. Build and run the app on this branch.
2. Run the reloadSettings gradle task.
3. Submit the report by running in a terminal: 
```
curl --location 'http://localhost:7071/api/reports?processing=async&allowDuplicate=false' \
--header 'client: ignore.ignore-mars-otc' \
--header 'Content-Type: application/hl7-v2' \
--header 'x-functions-key: 7E7BBE1C-BF96-40BA-A728-587FD8107133' \
--data 'MSH|^~\&|MMTC.PROD^2.16.840.1.113883.3.8589.4.2.106.1^ISO|CAREEVOLUTION^00Z0000024^CLIA|AIMS.INTEGRATION.PRD^2.16.840.1.114222.4.3.15.1^ISO|AIMS.PLATFORM^2.16.840.1.114222.4.1.217446^ISO|20240403205305+0000||ORU^R01^ORU_R01|20240403205305_dba7572cc6334f1ea0744c5f235c823e|P|2.5.1|||NE|NE|||||PHLabReport-NoAck^ELR251R1_Rcvr_Prof^2.16.840.1.113883.9.11^ISO
        SFT|CAREEVOLUTION|2022|MMTC.PROD|16498||20240402
        PID|1||8be6fa3710374dcebe0174e0fd5a1a7c^^^MMTC.PROD&2.16.840.1.113883.3.8589.4.2.106.1&ISO^PI||^^^^^^~^^^^^^||||||^^^^02139^USA||^^^^^111^1111111
        ORC|RE||^MMTC.PROD^2.16.840.1.113883.3.8589.4.2.106.1^ISO|||||||||^^||^^^^^^|||||||SA.OTCSelfReport|^^^^02139^^^^|^^^^^^
        OBR|1||^MMTC.PROD^2.16.840.1.113883.3.8589.4.2.106.1^ISO|94558-4^SARS-CoV-2 (COVID-19) Ag [Presence] in Respiratory specimen by Rapid immunoassay^LN^^^^2.71|||20240403120000-0400|||||||||^^|^^^^^^|||||20240403120000-0400|||F
        OBX|1|CWE|94558-4^SARS-CoV-2 (COVID-19) Ag [Presence] in Respiratory specimen by Rapid immunoassay^LN^^^^2.71||260373001^Detected^SCT^^^^20200901||||||F||||00Z0000042||BinaxNOW COVID-19 Antigen Self Test_Abbott Diagnostics Scarborough, Inc._EUA^^99ELR^^^^Vunknown||20240403120000||||SA.OTCSelfReport^^^^^&2.16.840.1.113883.3.8589.4.1.152&ISO^XX^^^00Z0000042|
        NTE|1|L|BinaxNOW COVID-19 Antigen Self Test_Abbott Diagnostics Scarborough, Inc._EUA
        OBX|2|NM|35659-2^Age at specimen collection^LN^^^^2.71||24|a^year^UCUM^^^^2.1|||||F||||00Z0000042||||||||SA.OTCSelfReport^^^^^&2.16.840.1.113883.3.8589.4.1.152&ISO^XX^^^00Z0000042||||||QST
        SPM|1|^dba7572cc6334f1ea0744c5f235c823e&MMTC.PROD&2.16.840.1.113883.3.8589.4.2.106.1&ISO||697989009^Anterior nares swab^SCT^^^^20200901|||||||||||||20240403120000-0400|20240403120000-0400'
```
4. Call the history API using the report ID in the previous step by running in a terminal:
```
curl --location 'http://localhost:7071/api/waters/report/<report_id_from_submission>/history' \
--header 'authentication-type: okta'
```
5. Verify that there are no errors.
<img width="1060" alt="image" src="https://github.com/user-attachments/assets/879a75a8-eb77-4947-b50d-20678c029058">

6. Submit the report by running in a terminal: 
```
curl --location 'http://localhost:7071/api/reports?processing=async&allowDuplicate=false' \
--header 'client: development.mars-otc-elr-onboarding-dev' \
--header 'Content-Type: application/hl7-v2' \
--header 'x-functions-key: 7E7BBE1C-BF96-40BA-A728-587FD8107133' \
--data 'MSH|^~\&|MMTC.PROD^2.16.840.1.113883.3.8589.4.2.106.1^ISO|CAREEVOLUTION^00Z0000024^CLIA|AIMS.INTEGRATION.PRD^2.16.840.1.114222.4.3.15.1^ISO|AIMS.PLATFORM^2.16.840.1.114222.4.1.217446^ISO|20240403205305+0000||ORU^R01^ORU_R01|20240403205305_dba7572cc6334f1ea0744c5f235c823e|P|2.5.1|||NE|NE|||||PHLabReport-NoAck^ELR251R1_Rcvr_Prof^2.16.840.1.113883.9.11^ISO
        SFT|CAREEVOLUTION|2022|MMTC.PROD|16498||20240402
        PID|1||8be6fa3710374dcebe0174e0fd5a1a7c^^^MMTC.PROD&2.16.840.1.113883.3.8589.4.2.106.1&ISO^PI||^^^^^^~^^^^^^||||||^^^^02139^USA||^^^^^111^1111111
        ORC|RE||^MMTC.PROD^2.16.840.1.113883.3.8589.4.2.106.1^ISO|||||||||^^||^^^^^^|||||||SA.OTCSelfReport|^^^^02139^^^^|^^^^^^
        OBR|1||^MMTC.PROD^2.16.840.1.113883.3.8589.4.2.106.1^ISO|94558-4^SARS-CoV-2 (COVID-19) Ag [Presence] in Respiratory specimen by Rapid immunoassay^LN^^^^2.71|||20240403120000-0400|||||||||^^|^^^^^^|||||20240403120000-0400|||F
        OBX|1|CWE|94558-4^SARS-CoV-2 (COVID-19) Ag [Presence] in Respiratory specimen by Rapid immunoassay^LN^^^^2.71||260373001^Detected^SCT^^^^20200901||||||F||||00Z0000042||BinaxNOW COVID-19 Antigen Self Test_Abbott Diagnostics Scarborough, Inc._EUA^^99ELR^^^^Vunknown||20240403120000||||SA.OTCSelfReport^^^^^&2.16.840.1.113883.3.8589.4.1.152&ISO^XX^^^00Z0000042|
        NTE|1|L|BinaxNOW COVID-19 Antigen Self Test_Abbott Diagnostics Scarborough, Inc._EUA
        OBX|2|NM|35659-2^Age at specimen collection^LN^^^^2.71||24|a^year^UCUM^^^^2.1|||||F||||00Z0000042||||||||SA.OTCSelfReport^^^^^&2.16.840.1.113883.3.8589.4.1.152&ISO^XX^^^00Z0000042||||||QST
        SPM|1|^dba7572cc6334f1ea0744c5f235c823e&MMTC.PROD&2.16.840.1.113883.3.8589.4.2.106.1&ISO||697989009^Anterior nares swab^SCT^^^^20200901|||||||||||||20240403120000-0400|20240403120000-0400'
```
7. Call the history API using the report ID in the previous step by running in a terminal:
```
curl --location 'http://localhost:7071/api/waters/report/<report_id_from_submission>/history' \
--header 'authentication-type: okta'
```
8. Verify errors are returned. 
![image](https://github.com/user-attachments/assets/b9e5dd2e-f4e9-4cb1-8418-3e2d7d65857f)


## Changes
- Added RADx Mars OTC Onboarding conformance profile files from NIST
- Moved existing RADx Mars OTC conformance profile files into a "production" folder since they are for the Production conformance profile
- Updated tests for moved file location
- Added validator that uses the RADx Mars OTC Onboarding conformance profile
- Added "mars-otc-elr-onboarding" topic that uses the new conformance profile validator
- Updated various locations to account for the new topic
- Added a sender and receiver to the development organization in organizations.yml that use the new topic
- Added sample_3.hl7 that will pass the validation for the RADx Mars OTC production, but will error on the onboarding
- Added a validator test

## Checklist

### Testing

## Linked Issues
- Fixes #15180
